mobilizon_reshare/publishers/coordinators/event_publishing/publish.py

```diff
@@ -40,18 +40,18 @@ class PublisherCoordinator(BaseEventPublishingCoordinator):
 
     def run(self) -> PublisherCoordinatorReport:
         errors = self._validate()
-        if errors:
-            return PublisherCoordinatorReport(
-                reports=errors, publications=self.publications
-            )
-
-        return self._publish()
+        successes = self._publish(errors)
+        return PublisherCoordinatorReport(
+            publications=self.publications,
+            reports=errors + successes
+        )
 
-    def _publish(self) -> PublisherCoordinatorReport:
+    def _publish(self, errors: Sequence[EventPublicationReport]) -> List[EventPublicationReport]:
         reports = []
-
+        publishers_with_errors = set(e.publication.publisher for e in errors)
```

Comment:
I would keep this logic in as a separate function, so that `_validate()` and `_publish` are not that coupled

Reply:
basically pass a clean list of publications to `_publish`, whose role is just to publish and not to filter and publish